core/lib/src/config/custom_values.rs

```diff
@@ -211,7 +211,7 @@ pub fn u64(conf: &Config, name: &str, value: &Value) -> Result<u64> {
 
 pub fn u16(conf: &Config, name: &str, value: &Value) -> Result<u16> {
     match value.as_integer() {
-        Some(x) if x >= 0 && x <= (u16::max_value() as i64) => Ok(x as u16),
+        Some(x) if x >= 0 && x <= (i64::from(u16::max_value())) => Ok(x as u16),
```

Comment:
Is there a reason to prefer this over the `as` coercion? The former seems clearer to me.

Reply:
I had the same question, this seems to be the answer as to why clippy does this lint:
> Rust’s `as` keyword will perform many kinds of conversions, including silently lossy conversions. Conversion functions such as `i32::from` will only perform lossless conversions. Using the conversion functions prevents conversions from turning into silent lossy conversions if the types of the input expressions ever change, and make it easier for people reading the code to know that the conversion is lossless.